backend/ethereum/subscription/resistanteventsub_test.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2021 - See NOTICE file for copyright holders.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package subscription_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/stretchr/testify/require"
+
+	"perun.network/go-perun/backend/ethereum/bindings/peruntoken"
+	"perun.network/go-perun/backend/ethereum/channel/test"
+	"perun.network/go-perun/backend/ethereum/subscription"
+	"perun.network/go-perun/log"
+	pctx "perun.network/go-perun/pkg/context"
+	pkgtest "perun.network/go-perun/pkg/test"
+)
+
+var event = func() *subscription.Event {
+	return &subscription.Event{
+		Name: "Approval",
+		Data: new(peruntoken.PerunTokenApproval),
+	}
+}
+
+// TestResistantEventSub_Confirmations tests that a TX is confirmed exactly
+// after being included in `confirmations` many blocks.
+func TestResistantEventSub_Confirmations(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+	rng := pkgtest.Prng(t)
+	require := require.New(t)
+	s := test.NewTokenSetup(ctx, t, rng)
+
+	for i := 0; i < 3; i++ {
+		finality := rng.Int31n(10) + 1
+		sub, err := subscription.Subscribe(ctx, s.CB, s.Contract, event, 0, uint64(finality))
+		require.NoError(err)
+		log.Debugf("Needed confirmations: %d", finality)
+		// Send and Confirm the TX. The simulated backend already mined a block here,
+		// so the TX has 1 confirmation now.
+		s.ConfirmTx(s.IncAllowance(ctx), true)
+		// Wait `finality-1` blocks.
+		for j := int32(0); j < finality-1; j++ {
+			NoEvent(require, sub)
+			log.Debug("Commit")
+			s.SB.Commit()
+		}
+		log.Debug("Waiting for confirm")
+		OneEvent(require, sub)
+		sub.Close()
+		s.SB.Commit() // Make sure the next EventSub does not get some old block.
+	}
+}
+
+// TestResistantEventSub_FinalityDepth tests that a TX is confirmed exactly
+// after `finalityDepth` blocks even when reorgs occur.
```

Comment:
```suggestion
// TestResistantEventSub_Reorg tests that a TX is confirmed exactly
// after `finalityDepth` blocks even when reorgs occur.
```

Also: Here you call it `finalityDepth`, above you call it confirmations. Either one or the other should be consistently used. I prefer `finalityDepth`.

Reply:
I renamed it everywhere.